common/src/HttpMessageHelper.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+#if NET451
+using System.Net.Http.Formatting;
+#endif
+
+namespace Microsoft.Azure.Devices.Shared
+{
+    internal static class HttpMessageHelper
+    {
+#if !NET451
+        private const string ApplicationJson = "application/json";
+#endif
+
+#if NET451
+        private static readonly JsonMediaTypeFormatter s_jsonFormatter = new JsonMediaTypeFormatter();
+#endif
+
+        internal static void SetHttpRequestMessageContent<T>(HttpRequestMessage requestMessage, T entity)
+        {
+#if NET451
+            requestMessage.Content = new ObjectContent<T>(entity, s_jsonFormatter);
+#else
+            string str = JsonConvert.SerializeObject(entity);
+            requestMessage.Content = new StringContent(str, Encoding.UTF8, ApplicationJson);
+#endif
+        }
+
+        internal static async Task<T> ReadHttpResponseMessageContentAsync<T>(HttpResponseMessage message, CancellationToken token)
```

Comment:
Elsewhere spelled out as `cancellationToken` but here just `token`. Use the full one for consistency?

Reply:
sure, looks like I missed it here